pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -778,36 +778,32 @@ cdef class _Timedelta(timedelta):
 
         if isinstance(other, _Timedelta):
             ots = other
-        elif PyDelta_Check(other) or isinstance(other, Tick):
+        elif (is_timedelta64_object(other) or PyDelta_Check(other)
```

Comment:
Any performance implications of checking this first now?

Reply:
when we have a timedelta64 object we're avoiding a few unnecessary checks below (L784 of which occurs in python-space).  otherwise we are delaying everything by the duration of a `is_timedelta64_object` call, which is small